org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/general/completion/FilePathCompletionParticipant.java

```diff
@@ -0,0 +1,235 @@
+/*******************************************************************************
+* Copyright (c) 2019 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.lsp4xml.extensions.general.completion;
+
+import static org.eclipse.lsp4xml.utils.StringUtils.getNormalizedPath;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.net.URI;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import javax.xml.crypto.dsig.XMLObject;
+
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.CompletionItemKind;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.lsp4xml.commons.BadLocationException;
+import org.eclipse.lsp4xml.dom.DOMDocument;
+import org.eclipse.lsp4xml.services.extensions.ICompletionParticipant;
+import org.eclipse.lsp4xml.services.extensions.ICompletionRequest;
+import org.eclipse.lsp4xml.services.extensions.ICompletionResponse;
+import org.eclipse.lsp4xml.settings.SharedSettings;
+import org.eclipse.lsp4xml.utils.CompletionSortTextHelper;
+import org.eclipse.lsp4xml.utils.StringUtils;
+
+/**
+ * FilePathCompletionParticipant
+ */
+public class FilePathCompletionParticipant implements ICompletionParticipant {
+
+	public static final String FILE_SCHEME = "file";
+
+	private static final FilenameFilter xsdFilter = new FilenameFilter() {
+
+		@Override
+		public boolean accept(File dir, String name) {
+			if(name.matches(".+\\.[^\\s]+$") == false) {
```

Comment:
It should be more performant to compile the pattern in Pattern class (like XMLScanner)